--- da_drventry.c.orig	2020-09-07 13:44:08.000000000 +0900
+++ da_drventry.c	2022-12-07 11:07:42.187610885 +0900
@@ -267,9 +267,11 @@
 int gpg_da_MINOR_f(void *file)
 {
 #if (LINUX_VERSION_CODE < VERSION(4,9,0))
-	return MINOR(((struct file *)file)->f_dentry->d_inode->i_rdev);
+//	return MINOR(((struct file *)file)->f_dentry->d_inode->i_rdev);
+	return iminor(file_inode((struct file *) file));
 #else
-	return MINOR(((struct file *)file)->f_path.dentry->d_inode->i_rdev);
+//	return MINOR(((struct file *)file)->f_path.dentry->d_inode->i_rdev);
+	return iminor(file_inode((struct file *) file))
 #endif
 }
 
@@ -421,6 +423,11 @@
 	return;
 }
 
+inline long interruptible_sleep_on_timeout(wait_queue_head_t * q, long timeout)
+{
+	return wait_event_interruptible_timeout(*q, 0, timeout);
+}
+
 /************************************************/
 /* gpg_da_interruptible_sleep_on_timeout        */
 /************************************************/
